File: 2- Regression/Support vector Regression (SVR)/python/SVR.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Tue Mar 28 17:20:35 2017

SVR - Support Vector Machine for regression

@author: Ilaria
"""


# Data Preprocessing Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

#we don't apply training and test set because we don't have enough data
# Splitting the dataset into the Training set and Test set
"""from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)
"""

# Feature Scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
sc_y = StandardScaler()
X = sc_X.fit_transform(X)
y= sc_y.fit_transform(y)

#fitting  SVR model to the dataset
from sklearn.svm import SVR
regressor = SVR(kernel = 'rbf')
regressor.fit(X,y)




#Predicting a new result with SVR
y_pred = sc_y.inverse_transform(regressor.predict(sc_X.transform(np.array([[6.5]]))))


#visualize SVR results
plt.scatter(X, y, color= 'red')
plt.plot(X, regressor.predict(X), color = 'blue')
plt.title('Reality vs Bluff(SVR regression)')
plt.xlabel('Position label')
plt.ylabel('Salary')
plt.show()

File: 2- Regression/Support vector Regression (SVR)/R/SVR.R
#SVR
#import data 
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]


#let's split between train and test set
#library(caTools)
#set.seed(123)
##true goes to training, false to test
#split= sample.split(dataset$Profit, SplitRatio = 0.8)
#training_set = subset(dataset, split == TRUE)
#test_set = subset(dataset, split == FALSE)

#fitting SVR
library(e1071)
regressor = svm(formula = Salary ~., 
                data = dataset, 
                type = 'eps-regression')

#predict a new result with SVR
y_pred = predict(regressor, data.frame(Level=6.5))

#visualize SVR regression 
library(ggplot2)
ggplot()+ 
  geom_point(aes(x = dataset$Level, y = dataset$Salary), colour = 'blue') + 
  geom_line(aes(x = dataset$Level, y = predict(regressor, newdata = dataset)), colour = 'blue') + 
  ggtitle('Truth vs bluff polynomial regression') + 
  xlab ('level') + 
  ylab('salary')





